src/thunderbots/software/ai/hl/stp/tactic/cherry_pick_tactic.cpp

```diff
@@ -0,0 +1,44 @@
+#include "ai/hl/stp/tactic/cherry_pick_tactic.h"
+
+#include "ai/hl/stp/action/move_action.h"
+#include "geom/util.h"
+
+CherryPickTactic::CherryPickTactic(const Rectangle& target_region, bool loop_forever)
+    : pass_generator(0.0), target_region(target_region), Tactic(loop_forever)
+{
+    pass_generator.setTargetRegion(target_region);
+}
+
+std::string CherryPickTactic::getName() const
+{
+    return "Cherry Pick Tactic";
+}
+
+void CherryPickTactic::updateParams(const World& world)
+{
+    this->world = world;
+}
+
+double CherryPickTactic::calculateRobotCost(const Robot& robot, const World& world)
+{
+    // Prefer robots closer to the target region
+    return dist(robot.position(), target_region);
+}
+
+std::unique_ptr<Intent> CherryPickTactic::calculateNextIntent(
+    intent_coroutine::push_type& yield)
+{
+    MoveAction move_action = MoveAction();
+    do
+    {
+        // Move the robot to be the best possible receiver for the best pass we can
+        // find (within the target region)
+        std::optional<AI::Passing::Pass> best_pass = pass_generator.getBestPassSoFar();
+        if (best_pass)
```

Comment:
Would this `best_pass` ever be a `nullopt`? If `best_pass` is `nullopt` this function will get stuck in an infinite loop. Is there a reasonable fallback if `best_pass` resolves to `false`? Maybe just use the previous position?

Also, we should probably add hysteresis to the position being moved to.

Reply:
Theoretically it shouldn't, but you're totally right that we should deal with case explicitly anyhow. Changed the `while` conditional to 
```
    } while (!move_action.done() && best_pass);
```

With regards to hysteresis, are there some other examples I could take a look at of how we've done it, so we're being consistent? Or should I just come up with a semi-general scheme?

Reply:
semi-general is fine. We don't have anything like this that I'm aware of right now.

Reply:
After discussion on slack, decided to follow the principle of [YAGNI](https://www.martinfowler.com/bliki/Yagni.html) and leave this out until there is demonstrable need for it.